### ISSUE TITLE
In NodeInfo (Model storage) pods should be keyed with both name and namespace.

### ISSUE THREAD

**Issue by mwielgus** [kubernetes/heapster#517]
The issue is present in the old code as well as in #508

**Comment by afein** [comment]
Node pods are currently keyed by concatenating the namespace and the pod name such as `default/frontend-xkyz`. 

This assures no pod name collisions can occur, but it is not the cleanest way of exposing that information. A PodPath struct can be created to hold the pod's namespace and its name.

**Comment by mwielgus** [comment]
In updatePod (#508):
```go
	// Get Pod pointer
	pod_ptr := rc.addPod(pod.Name, pod.UID, namespace, node)
```
Does pod.Name contain in fact namespce + name?

**Comment by afein** [comment]
actually you are right, pod.Name is the pod's name, as provided by the PodElement struct that is consumed from the cache.

The GetNodePods method manually constructs the `<namespace>/<podname>` path by calling the findPodNamespace method. Instead of that, we could directly key the NodeInfo.Pods map by `<namespace>/<podname>`

**Comment by mwielgus** [comment]
I will have the fix in PR for #513 (which is based on #508).

**Comment by mwielgus** [comment]
Fixed in #522
